src/lib/MVC/Symfony/EventListener/SiteAccessMatchListener.php

```diff
@@ -193,9 +194,14 @@ private function buildMatcherFromSerializedClass(
         string $matcherClass,
         string $serializedMatcher
     ): SiteAccess\Matcher {
-        $matcher = null;
         if ($this->siteAccessMatcherRegistry->hasMatcher($matcherClass)) {
             $matcher = $this->siteAccessMatcherRegistry->getMatcher($matcherClass);
+            $matcher = $serializer->deserialize(
+                $serializedMatcher,
+                $matcherClass,
+                'json',
+                [AbstractNormalizer::OBJECT_TO_POPULATE => $matcher]
```

Comment:
This in the context of L198 looks a bit weird. I'd just inline getMatcher instead of creating a variable and then overriding it right away.
```suggestion
            $matcher = $this->siteAccessMatcherRegistry->getMatcher($matcherClass);
            $matcher = $serializer->deserialize(
                $serializedMatcher,
                $matcherClass,
                'json',
                [
                    AbstractNormalizer::OBJECT_TO_POPULATE => $this->siteAccessMatcherRegistry->getMatcher($matcherClass)
                ]
```

POV ping @Steveb-p